components/automate-ui/src/app/entities/infra-roles/infra-role.requests.ts

```diff
@@ -23,4 +28,11 @@ export class InfraRoleRequests {
     return this.http.get<InfraRole>(
         `${env.infra_proxy_url}/servers/${server_id}/orgs/${org_id}/roles/${name}`, {headers});
   }
+
+  public getRoleSearch(payload: RoleSearchPayload): Observable<RoleSearchResponse> {
+    const params = `search_query.q=name:${payload.roleId}*&search_query.start=${payload.page}&search_query.row=${payload.per_page}`;
```

Comment:
@msorens  Yes, we are using an asterisk to show all records or similar records.
"*" is used here to wild the search 
we have to search the similar records
for more detail: https://docs.chef.io/workstation/knife_search/

Reply:
That seems odd for an ID; when would an ID not be exact?

Reply:
As currently, the search box is used to search by name
It seems there might be confusion because of the variable name -` roleID`, I'll rename it to `roleName`.

Reply:
Yes, please do. 😁 And let's change this line of code to be more transparent:
```
    const nameTarget = payload.roleId + '*'; // Add asterisk to do wildcard search
    const params = `search_query.q=name:${nameTarget}&search_query.start=${payload.page}&search_query.row=${payload.per_page}`;
```

Reply:
Sure will do that. :)